src/pages/contact/contact.ts

```diff
@@ -1,15 +1,58 @@
 import { Component } from "@angular/core";
+import { AuthHttp } from "angular2-jwt";
+import { Pipe } from "@angular/core";
 
+import { ModalController } from "ionic-angular";
 import { NavController } from "ionic-angular";
 
+import { environment } from "../../environments/environment";
+import { ContactModalPage } from "../modals/contact-modals/contact-modal";
+
 @Component({
   selector: "page-contact",
   templateUrl: "contact.html",
 })
 export class ContactPage {
 
-  constructor(public navCtrl: NavController) {
+  public contacts: any;
+
+  constructor(public navCtrl: NavController,
+              public modalCtrl: ModalController,
+              public auth_http: AuthHttp) {
+    this.auth_http.get(environment.server_url + "/api/user/contact")
+        .map((res) => res.json())
+        .subscribe((result) => this.contacts = result.data);
+  }
 
+  showContactModal(contact) {
+    let modal = this.modalCtrl.create(ContactModalPage, { contact: contact });
+    modal.present();
   }
 
 }
+
+@Pipe({
+  name: 'phone'
+})
+export class PhonePipe {
+  /**
+   * Modifies the input telephone number to match standard representations.
+   *
+   * @param val Telephone number as string.
+   * @param args
+   * @returns {String}
+   */
+  transform(val, args) {
+    if (val.length == 7) return val.slice(0, 3) + '-' + val.slice(3, 7);
+
```

Comment:
I think the if statement would be more clear if the return was on a different line.